native-schema-registry/src/main/java/com/amazonaws/services/schemaregistry/ExceptionWriter.java

```diff
@@ -30,4 +35,48 @@ private static void writeException(String message, C_GlueSchemaRegistryErrorPoin
         throwError(errorPointer, errorMessageAsCString.get(), FAILURE_CODE);
         errorMessageAsCString.close();
     }
+    /**
+     * Captures the full stack trace message of a Throwable, so it can be observed in the target language layer,
+     * For example, an exception thrown in Java layer will have these messages printed in C# layer:
+     *
+        Exception occurred while fetching or registering schema definition = {"type":"record","name":"User","namespace":"example.avro","fields":[{"name":"name","type":"string"},{"name":"favorite_number","type":["int","null"]},{"name":"favorite_color","type":["string","null"]}]}, schema name = test-topic-json
+            com.amazonaws.services.schemaregistry.common.SchemaByDefinitionFetcher.getORRegisterSchemaVersionId(SchemaByDefinitionFetcher.java:99)
+            com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializationFacade.getOrRegisterSchemaVersion(GlueSchemaRegistrySerializationFacade.java:86)
+            com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializationFacade.encode(GlueSchemaRegistrySerializationFacade.java:125)
+            com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializerImpl.encode(GlueSchemaRegistrySerializerImpl.java:79)
+            com.amazonaws.services.schemaregistry.GlueSchemaRegistrySerializationHandler.encodeWithSchema(GlueSchemaRegistrySerializationHandler.java:90)
+        at AWSGsrSerDe.GlueSchemaRegistrySerializer.Encode(String transportName, GlueSchemaRegistrySchema schema, Byte[] bytes) in /Users/haoqyang/Documents/gsr_multilang/aws-glue-schema-registry/native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/GlueSchemaRegistrySerializer.cs:line 78
+        at AWSGsrSerDe.serializer.GlueSchemaRegistryKafkaSerializer.Serialize(Object data, String topic) in /Users/haoqyang/Documents/gsr_multilang/aws-glue-schema-registry/native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/serializer/GlueSchemaRegistryKafkaSerializer.cs:line 68
+        at AWSGsrSerDe.Tests.serializer.GlueSchemaRegistryKafkaSerializerTests.KafkaSerDeTestForAvroGenericRecord() in /Users/haoqyang/Documents/gsr_multilang/aws-glue-schema-registry/native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe.Tests/serializer/GlueSchemaRegistryKafkaSerializerTests.cs:line 52
+     */
+    private static String getFullStackTraceMessage(Throwable throwable) {
+        String throwableMessage;
+        StringJoiner messageJoiner = new StringJoiner("\n");
+        Boolean shouldAppendStackTrace = false;
+
+        if (throwable == null) {
+            throwableMessage = "NULL is thrown";
+        } else if (throwable.getMessage() == null) {
+            throwableMessage = throwable.getClass().getName() + " is thrown, but there is no detail message";
+        } else if (throwable.getStackTrace() == null) {
+            throwableMessage = throwable.getMessage() + "\n but Stacktrace is NULL";
```

Comment:
No need of "but Stack is NULL". It's confusing whether it's part of the exception.

Reply:
Fixed